src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,63 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
 
+    public void push(T value) {
+        Node<T> newElement = new Node<>(value, null);
+        if (size == 0) {
+            tail = newElement;
+            head = newElement;
+        } else {
+            tail.next = newElement;
+            tail = newElement;
+        }
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkException();
+        return tail.item;
     }
 
     public T pop() {
-        return null;
+        checkException();
+        Node<T> oldElement = tail;
+        tail = findElement();
```

Comment:
Isn't it too complex? You update tail each time you push new value

Reply:
I fixed this